### ISSUE TITLE
Authorization grant server not reachable at "https://homebridge.local:8582/"! Choose a port which is reachable from within this network.

### ISSUE THREAD

**Issue by Tiliii** [QuickSander/homebridge-mieleathome#46]
Hey guys,
I really need help. I'm a bloody beginner with Homebridge and so on, I got some basic knowledge about this whole theme but its really basic. I'm trying to get this plugin to work but it fails every single time. At first I didn't got to the authentication process where you have to put in your miele@home credentials. I then downgraded the version. Then I got to the authentication process, I put my login credentials in and clicked authenticate. Then the browser times out and nothing happens. After a minute or something like that the plugin says nothing and Homebridge wants me to restart it. After the restart there is no real error code everything loads up but Homebridge isn't showing any new products. After I update the plugin and restart Homebridge I get this error code in the Homebridge plugin settings "Authorization grant server not reachable at "https://homebridge.local:8582/"! Choose a port which is reachable from within this network.". I don't really know what I should do now, I tried to take another port but then it gives me other error codes. In the Homebridge protocol I get this red error: "Miele API request https://api.mcs3.miele.com/v1/devices?language=en failed with status 401: "Unauthorized".". Would be great if somebody could help me. 

I'm using a raspberry pi3 with the Homebridge os installed. 
the raspberry is connected via ethernet with my network. 
Node.js Version: v14.17.3
Nem version: v6.14.13

[15/07/2021, 20:43:23] Got SIGTERM, shutting down Homebridge...
[15/07/2021, 20:43:28] [Hue] goodbye
[15/07/2021, 20:43:28] [HB Supervisor] Homebridge Process Ended. Code: 143, Signal: null
[15/07/2021, 20:43:33] [HB Supervisor] Restarting Homebridge...
[15/07/2021, 20:43:33] [HB Supervisor] Starting Homebridge with extra flags: -I
[15/07/2021, 20:43:33] [HB Supervisor] Started Homebridge v1.3.4 with PID: 10312
Initializing HAP-NodeJS v0.9.4...
[15/07/2021, 20:43:34] Loaded config.json with 0 accessories and 3 platforms.
[15/07/2021, 20:43:34] Loaded 0 cached accessories from cachedAccessories.
[15/07/2021, 20:43:34] ---
[15/07/2021, 20:43:35] Loaded plugin: homebridge-config-ui-x@4.41.1
[15/07/2021, 20:43:35] Registering platform 'homebridge-config-ui-x.config'
[15/07/2021, 20:43:35] ---
[15/07/2021, 20:43:37] Loaded plugin: homebridge-hue@0.13.17
[15/07/2021, 20:43:37] Registering platform 'homebridge-hue.Hue'
[15/07/2021, 20:43:37] ---
[15/07/2021, 20:43:37] Loaded plugin: homebridge-mieleathome@3.3.0
[15/07/2021, 20:43:37] Registering platform 'homebridge-mieleathome.Miele@home'
[15/07/2021, 20:43:37] ---
[15/07/2021, 20:43:37] Loading 3 platforms...
[15/07/2021, 20:43:37] [Config] Initializing config platform...
[15/07/2021, 20:43:37] [Config] Running in Service Mode
[15/07/2021, 20:43:37] [Hue] Initializing Hue platform...
[15/07/2021, 20:43:37] [Hue] homebridge-hue v0.13.17, node v14.17.3, homebridge v1.3.4
[15/07/2021, 20:43:37] [Hue] warning: recommended version: node v14.17.2 LTS
[15/07/2021, 20:43:37] [Miele@home] Initializing Miele@home platform...
[15/07/2021, 20:43:37] [Miele@home] Token loaded from persistent storage.
[15/07/2021, 20:43:37] [Hue] Philips hue: Signify Netherlands B.V. BSB002 bridge v1945163030, api v1.45.0
[15/07/2021, 20:43:37] [Miele@home] Miele API request https://api.mcs3.miele.com/v1/devices?language=en failed with status 401: "Unauthorized".

**Comment by QuickSander** [comment]
Hi,
The plugin starts a server when you open settings. This server must be reachable. If not, the plugin cannot accept an incoming request from Miele.

Is there anything special about your network configuration? 

You can try the manual token generation process as described on the wiki of this GitHub repository.

The reason you get unauthorised is because somehow you were successful in creating a token in the past, but it apparently became invalid. Th plugin will always try to auto-refresh the token if Homebridge is kept running.

Please try the manual token retrieval process.

**Comment by QuickSander** [comment]
I see you are running homebridge using https. The server it starts is not using https, but nevertheless tries to contact https.

**Comment by QuickSander** [comment]
Do you have logging of the moment you open the setup?

**Comment by QuickSander** [comment]
No response, assume solved after update.

**Comment by ixs** [comment]
Okay, I think I have the same issue. Maybe some more detail here:
Running the docker container `oznu/homebridge` with host networking enabled.

Server log says:
`[3/8/2022, 6:45:58 PM] [Homebridge UI] [homebridge-mieleathome] Temporary authorization grant receiver server running on http://0.0.0.0:8582`

But then, on the authorization grant screen, I am getting the following message:
> Authorization grant server not reachable at "http://diskstation.local:8582/"! Choose a port which is reachable and available from within this network.

I can easily access "http://diskstation.local:8582/" in the browser though and see a "Resource "/" not found." error message.

So pretty much what @Tiliii was seeing as well. The weird thing however is, that after about 10min of doing something else, the Authorization server was shown as reachable and authorization succeeded.
No idea why... But maybe it helps you understand what might be going on?

**Comment by QuickSander** [comment]
Thanks, I'm looking into it to see if at least I can change the code a bit to be more robust (since I have no clue yet where the cause is)

**Comment by QuickSander** [comment]
Could you try if it helps to change the port from 8582 to something else and then back to 8582?
Editing that box will retrigger the grant server testing procedure.

**Comment by Flashchip75** [comment]
Hi, I still have the same problem as the two described before. Is there a new approach on this issue? I did all that was mentioned before and tried to fix it my self but none of these worked

**Comment by na-ma-ste** [comment]
I have been facing the same problem for a few days now. Any Ideas?

**Comment by na-ma-ste** [comment]
Hi, I think the plugin is not able to read the IP address of the Pi. I get the following error although I have a regular IP address in Homebridge

[homebridge-mieleathome] Authoriation grant server error: EADDRINUSE Error: listen EADDRINUSE: address already in use 0.0.0.0:80

**Comment by QuickSander** [comment]
Seems you use port 80 which is the default http server port and thus highly likely to be in use already. Did you try another port number?

**Comment by na-ma-ste** [comment]
yes, no effect. But deactivating and activating the plugin helped.

**Comment by vcesinger** [comment]
for me it worked when I opened the homebridge.local with http:// instead of https://. - I would assume its a problem with the certificate when trying to access

**Comment by kaduffey** [comment]
using http:// on a V1.11 Homebridge using a Orange Pi Zero2W dedicated to HB via Node.js 24.13; plugin v3.5.0

Create my id/key. Using port 8582 (switched it back and forth); also tried disable/restart/enable/restart

[1/30/2026, 9:21:39 PM] [Miele@home] Miele API request https://api.mcs3.miele.com/v1/devices?language= failed with status 401: "Unauthorized".

[1/30/2026, 9:24:00 PM] [Homebridge UI] Terminal session ended.
[1/30/2026, 9:24:44 PM] [Homebridge UI] [homebridge-mieleathome] Incoming Request: /is_token_stored
[1/30/2026, 9:24:44 PM] [Homebridge UI] [homebridge-mieleathome] No valid token in persistent storage. Setup required. Token in storage: {"expires_in":3600,"creation_date":"2026-01-31T05:21:38.950Z"}
[1/30/2026, 9:24:44 PM] [Homebridge UI] [homebridge-mieleathome] Incoming Request: /initialize_server
[1/30/2026, 9:24:44 PM] [Homebridge UI] [homebridge-mieleathome] Generated state: c05acc323b21fd5bd14e2ceee2ee89fb
[1/30/2026, 9:24:44 PM] [Homebridge UI] [homebridge-mieleathome] Temporary authorization grant receiver server running on http://0.0.0.0:8582

Plugin config returns: Authorization grant server not reachable at "http://xome.local:8582/"! Choose a port which is reachable and available from within this network.

Json is simple:
{
            "clientID": "xxxxxxx",
            "clientSecret": "yyyyyyyyyyy",
            "platform": "Miele@home"
        }
